### ISSUE TITLE
pip install raven 6.2.0 is missing contextlib2

### ISSUE THREAD

**Issue by arthurlogilab** [getsentry/raven-python#1086]
We are getting the following error on our jenkins : 

```
[snip]
raven/__init__.py:54: in <module>
    from raven.base import * 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    """
    
    from __future__ import absolute_import
    
    import zlib
    import logging
    import os
    import sys
    import time
    import uuid
    import warnings
    
    from datetime import datetime
    from inspect import isclass
    from random import Random
    from types import FunctionType
    from threading import local
    
    if sys.version_info >= (3, 2):
        import contextlib
    else:
>       import contextlib2 as contextlib
E       ImportError: No module named contextlib2
```

it seems to have appeared since the release of 6.2.0. 

pip install raven in a clean virtualenv doesn't seem to be pulling contextlib2...

**Comment by arthurlogilab** [comment]
We're talking about py27 by the way.

**Comment by xeor** [comment]
This just broke my build as well :)
Adding `contextlib2` solved it..

py27

**Comment by superseed** [comment]
It seems like it's a problem with the way the wheel is built:
```
$ pip install --no-binary :all: raven
Collecting raven
  Using cached https://pypi.pv.ocshq.com/root/pypi/+f/775/95056f1b7491b/raven-6.2.0.tar.gz
Collecting contextlib2 (from raven)
  Using cached https://pypi.pv.ocshq.com/root/pypi/+f/95f/e420941ebc649/contextlib2-0.5.5.tar.gz
Skipping bdist_wheel for raven, due to binaries being disabled for it.
Skipping bdist_wheel for contextlib2, due to binaries being disabled for it.
Installing collected packages: contextlib2, raven
  Running setup.py install for contextlib2 ... done
  Running setup.py install for raven ... done
Successfully installed contextlib2-0.5.5 raven-6.2.0
```
```
$ pip install raven
Collecting raven
  Using cached https://pypi.pv.ocshq.com/root/pypi/+f/a74/bf5a1d2ad4eef/raven-6.2.0-py2.py3-none-any.whl
Installing collected packages: raven
Successfully installed raven-6.2.0
```

**Comment by kdeldycke** [comment]
I guess the package shouldn't be [declared as a universal wheel](https://github.com/getsentry/raven-python/blob/6.2.0/setup.cfg#L15-L16), because [dependencies are computed dynamically](https://github.com/getsentry/raven-python/blob/6.2.0/setup.py#L57-L64).

Maybe you should use extra Python requirements, as in: https://github.com/kdeldycke/meta-package-manager/blob/v2.6.0/setup.py#L49

**Comment by ashwoods** [comment]
@kdeldycke @superseed @arthurlogilab could you please check if 6.2.1 fixes the issue for you?

still checking out why this worked locally for me.

**Comment by Jnesselr** [comment]
Running a python setup.py install on master currently works for me. It also works on the fix-contextlib-wheel branch. Installing 6.2.1 from pip does seem to install contextlib2, though.

**Comment by superseed** [comment]
Yep, seems like it's fixed !
```
pip install raven
Collecting raven
  Downloading raven-6.2.1-py2.py3-none-any.whl (285kB)
    100% |████████████████████████████████| 286kB 2.8MB/s
Collecting contextlib2; python_version < "3.2" (from raven)
  Downloading contextlib2-0.5.5-py2.py3-none-any.whl
Installing collected packages: contextlib2, raven
Successfully installed contextlib2-0.5.5 raven-6.2.1
```

**Comment by ashwoods** [comment]
Still not perfect though. bdist_wheel will fail for setuptools<18. Permanent fix coming up in 6.2.2 ;)

**Comment by superseed** [comment]
I think it's really what @kdeldycke said earlier: when the wheel is built, the dependencies are calculated with whatever `sys.version_info` happens to return at build time. Which means building the wheel with a python2 will yield a wheel where the package depends on contextlib2, and building it with python3 will yield a wheel without this dependency.

Maybe you usually uploaded a wheel built with python2, but this timed used python3 ?
